src/utils/paymentQRHelper.test.ts

```diff
@@ -49,70 +49,123 @@ describe("tests for findValueByKey", () => {
-            validationRegex: "[\\s\\S]*",
-            value:
-              "00020101021126520008com.grab0136c99990dc-ab89-442f-accb-1fd96dd4eff627330015sg.com.dash.www0110000000901628460010com.myfave0128https://myfave.com/qr/6ntdz329360009SG.AIRPAY0119936009180000000468751810007SG.SGQR01121809112DE3C7020701.000703065743700402010503105060400000708202012105204581253037025802SG5920UDDERS UPPER THOMSON6009Singapore63047AEC",
-            textInputType: "PAYMENT_QR",
-          },
-          {
-            label: "merchantName",
-            isOptional: true,
-            value: "",
-            validationRegex: "^[a-zA-Z0-9 ]*$",
-            textInputType: "STRING",
-          },
-          {
-            label: "propertyThatDoesNotExist",
-            isOptional: true,
-            value: "",
-            textInputType: "STRING",
-          },
+          { ...paymentQRIdentifierInput, value: paymentQRPayloadWithoutNETS },
+          { ...merchantNameIdentifierInput, value: "UDDERS UPPER THOMSON" },
         ],
       },
-    ];
+    ]);
   });
 
-  it("should return updated transactions with identifiers properly", () => {
+  it("should return updated transactions with identifiers properly if property does not exist", () => {
     expect.assertions(1);
     expect(
-      getUpdatedTransactionsPaymentQRIdentifiers(transactionsBeforeUpdate)
+      getUpdatedTransactionsPaymentQRIdentifiers([
+        {
+          category: "category-a",
+          quantity: 1,
+          identifierInputs: [
+            { ...paymentQRIdentifierInput, value: paymentQRPayloadWithoutNETS },
+            merchantIdIdentifierInput,
+          ],
+        },
+      ])
     ).toStrictEqual([
       {
         category: "category-a",
         quantity: 1,
         identifierInputs: [
-          {
-            label: "Payment QR",
-            scanButtonType: "QR",
-            validationRegex: "[\\s\\S]*",
-            value:
-              "00020101021126520008com.grab0136c99990dc-ab89-442f-accb-1fd96dd4eff627330015sg.com.dash.www0110000000901628460010com.myfave0128https://myfave.com/qr/6ntdz329360009SG.AIRPAY0119936009180000000468751810007SG.SGQR01121809112DE3C7020701.000703065743700402010503105060400000708202012105204581253037025802SG5920UDDERS UPPER THOMSON6009Singapore63047AEC",
-            textInputType: "PAYMENT_QR",
-          },
-          {
-            label: "merchantName",
-            isOptional: true,
-            value: "UDDERS UPPER THOMSON",
-            validationRegex: "^[a-zA-Z0-9 ]*$",
-            textInputType: "STRING",
-          },
-          {
-            label: "propertyThatDoesNotExist",
-            isOptional: true,
-            value: "",
-            textInputType: "STRING",
-          },
+          { ...paymentQRIdentifierInput, value: paymentQRPayloadWithoutNETS },
+          merchantIdIdentifierInput,
+        ],
+      },
+    ]);
+  });
+
+  it("should return updated transactions with identifiers properly for NETS", () => {
```

Comment:
would it be useful to add a test here to test that `it should return updated transactions with identifiers properly if property does not exist`?

Reply:
Great idea, it would make the behavior much clearer! Added in [1952a79](https://github.com/rationally-app/mobile-application/pull/532/commits/1952a794b3e8ceaf5aee40d0be10a090af0cf6ff)!

Reply:
Niceee